### PR TITLE
Users next page button fix

### PR DESCRIPTION
**Pull request by Asymmentric**
##### Fix for the issue https://github.com/llaske/sugarizer-server/issues/364

Complete: 
- Fixed 'Next page >>' and '<< Previous page' buttons on `dashboard/users`  which now considers the search criteria while displaying user details
- form on `/dashboard/users` has 'username' as query parameter, so replaced 'q' query parameters by 'username' throughout 

In Progress:
- 'Next page >>' for assignments

### CODE REVIEW

**Comment by NikhilM98** [comment]
Can you clean this PR?
- Get rid of commits which are not yours.
- Remove unnecessary comments and white space from the code.

**Comment by Asymmentric** [comment]
hi @NikhilM98 
i tried cleaning it. Please let me know if it can be improved

File: docs/migrate.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ First identify the id for the MongoDB container by running.
 docker ps
 ```
 
-The id is the one on the line named `sugarizer-server_server`.
+The id is the one on the line named `sugarizer-server_mongodb`.
```
Comment (NikhilM98) [inline docs/migrate.md:49]: Create a different PR to address this typo.

File: api/controller/users.js [hunk 2]
```diff
@@ -559,11 +554,10 @@ function addQuery(filter, params, query, default_val) {
 
 	//check default case
 	query = query || {};
-
 	//validate
 	if (typeof params[filter] != "undefined" && typeof params[filter] === "string") {
-
-		if (filter == 'q') {
+		
+		if(filter=='username'){
```
Comment (NikhilM98) [inline api/controller/users.js]: There is no point in replacing `q` with `username`. `q` is commonly used for search operations.

File: api/controller/users.js [hunk 3]
```diff
@@ -559,11 +554,10 @@ function addQuery(filter, params, query, default_val) {
 
 	//check default case
 	query = query || {};
-
```
Comment (NikhilM98) [inline api/controller/users.js:562]: These delete line changes are unnecessary.

File: api/controller/users.js [hunk 4]
```diff
@@ -45,8 +45,7 @@ exports.init = function(settings, database) {
  * @apiParam {String} id Unique user id
  * 
  * @apiSuccess {String} _id Unique user id
- * @apiSuccess {String} name Unique user name
- * @apiSuccess {String} role User role (student or admin)
+ * @apiSuccessearchs {String} role User role (student or admin)
  * @apiSuccess {Object} color Buddy color
```
Comment (NikhilM98) [inline api/controller/users.js:49]: What's the purpose of this change?

File: dashboard/controller/users/index.js [hunk 5]
```diff
@@ -71,7 +70,7 @@ exports.index = function(req, res) {
 
 				// get classrooms list
 				getClassrooms(req, function(classrooms){
-	
+
```
Comment (NikhilM98) [inline dashboard/controller/users/index.js:73]: The changes with extra whitespaces are unnecessary.

**Review by NikhilM98 — CHANGES_REQUESTED** [review]

**Comment by NikhilM98** [comment]
In this PR I can see that you have changed some variable names, but how does it fixes the next page button issue?

**Comment by Asymmentric** [comment]
Suppose when the SHOW RESULTS button is clicked it requests   `_dashboard/users?username=n&role=student_` here it was fetching the results good and it was being resolved as  `query['q']=req.query.username`  but the _next_page_ links which were being created by `formPaginatedUrl()` looked something like  `_dashboard/users?q=n&role=student&offset=10&limit=10_` 

The problem was due to `req.query.q` not being handled there so the search criteria was not being considered.

**Comment by NikhilM98** [comment]
I just tested the issue. I see what you mean. However, it may be better if the keys like `username`, `name`, and `classroom` are reserved for exact string matching. For searching with partial string matching, it may be better to use `q`.

What do you think about doing something like [this](https://github.com/NikhilM98/sugarizer-server/commit/642260896f1e882c7fbcd685c6b36851dbfee0c2)?

**Comment by Asymmentric** [comment]
> What do you think about doing something like [this](https://github.com/NikhilM98/sugarizer-server/commit/642260896f1e882c7fbcd685c6b36851dbfee0c2)?

True, that is a fair approach but rather than changing element names in the ejs file can this be a possible solution 

By handling _req.query.q_

`if (req.query.username != '' || req.query.q!='') {
 query['q'] = req.query.username || req.query.q;
 }`

`if (req.query.classroom != '' || req.query.q!='') {
 query['q'] = req.query.classroom || req.query.q;
 }`

**Comment by NikhilM98** [comment]
Yes this is a possible solution too, but it is prone to unexpected behavior, for example if in future both variables are defined. Also, it may be confusing to use two variables for the same parameter.

**Comment by Asymmentric** [comment]
Yes sure it might make it inconsistent

**Comment by NikhilM98** [comment]
Can you make your changes consistent with the suggested ones and squash your commits

**Comment by Asymmentric** [comment]
@NikhilM98 I've made the changes. Let know if any improvements

**Comment by NikhilM98** [comment]
There are several changes in this PR which are not relevant to this issue: [changes](https://github.com/llaske/sugarizer-server/pull/379/files).

Can you remove the unnecessary changes and squash your commits into one or two commits?

**Comment by kartikaysaxena** [comment]
Can I contribute to the same?

**Comment by NikhilM98** [comment]
@kartikaysaxena the solution is already finalized and coded. This tasks doesn't require collaboration between multiple contributions for now. Feel free to experiment with Sugarizer Server and report any issues you find in the application.

**Comment by llaske** [comment]
@Asymmentric could you close one of the two PR?

**Pull request closed without merging**